modern/src/common/components/SelectField.jsx

```diff
@@ -33,20 +40,36 @@ const SelectField = ({
   if (items) {
     return (
       <FormControl fullWidth={fullWidth}>
-        <InputLabel>{label}</InputLabel>
-        <Select
-          label={label}
-          multiple={multiple}
-          value={value}
-          onChange={onChange}
-        >
-          {!multiple && emptyValue !== null && (
-            <MenuItem value={emptyValue}>{emptyTitle}</MenuItem>
-          )}
-          {items.map((item) => (
-            <MenuItem key={keyGetter(item)} value={keyGetter(item)}>{titleGetter(item)}</MenuItem>
-          ))}
-        </Select>
+        {multiple ? (
+          <>
+            <InputLabel>{label}</InputLabel>
+            <Select
+              label={label}
+              multiple
+              value={value}
+              onChange={onChange}
+            >
+              {items.map((item) => (
+                <MenuItem key={keyGetter(item)} value={keyGetter(item)}>{titleGetter(item)}</MenuItem>
+              ))}
+            </Select>
+          </>
+        ) : (
+          <>
+            <Autocomplete
+              size="small"
+              options={items}
+              getOptionLabel={getOptionLabel}
+              renderOption={(props, option) => (
+                <MenuItem {...props} key={keyGetter(option)} value={keyGetter(option)}>{titleGetter(option)}</MenuItem>
+              )}
+              isOptionEqualToValue={(option, value) => keyGetter(option) == value}
+              value={value}
+              onChange={(e, nV) => {onChange({target:{value:nV ? keyGetter(nV) : emptyValue}})}}
```

Comment:
Can you please rename `nV` to something else. Maybe just `value`.

Also please add spacings to the object. Maybe splitting object into multiple lines is better to make it more readable.

Reply:
What do you think about reworking this and always returning just the value? And of-course fixing all the callbacks. From what I can see, absolutely all just do `e.target.value` and nothing else, so I think it would be a lot cleaner to just return the value and be done with it. Ofcourse I would wrap the `Select` callback too, for consistency.

Reply:
I kind of like the fact that we can replace back with the standard select if needed. It also consistent with all other input fields.

Reply:
Ok, so I removed the fragment and for now just added spaces and it looks readable to me just ok - please check and tell me if I need to break it in more lines.